lib/slack-ruby-bot/app.rb

```diff
@@ -3,26 +3,36 @@
 module SlackRubyBot
   class App < Server
     def initialize(options = {})
-      SlackRubyBot.configure do |config|
-        config.token = ENV['SLACK_API_TOKEN'] || raise("Missing ENV['SLACK_API_TOKEN'].")
-        config.aliases = ENV['SLACK_RUBY_BOT_ALIASES'].split(' ') if ENV['SLACK_RUBY_BOT_ALIASES']
-      end
-      Slack.configure do |config|
-        config.token = SlackRubyBot.config.token
-      end
-      super
+      server_options = options.select { |k, _| k.to_sym == :token }
+      super(server_options)
     end
 
     def config
       SlackRubyBot.config
     end
 
     def self.instance
-      @instance ||= new
+      @instance ||= begin
+        configure!
+        new(token: SlackRubyBot.config.token)
+      end
     end
 
     private
 
+    def self.configure!
+      SlackRubyBot.configure do |config|
+        config.token ||= ENV['SLACK_API_TOKEN']
+        raise("Missing ENV['SLACK_API_TOKEN'].") if config.token.nil?
+
+        config.aliases = ENV['SLACK_RUBY_BOT_ALIASES'].split(' ') if ENV['SLACK_RUBY_BOT_ALIASES']
```

Comment:
`ENV.key?('SLACK_RUBY_BOT_ALIASES')` maybe?

Generally on whether to check for `.nil?` (like here) or whether it's set: I am of the opinion that someone can do `config.token = nil` and that is _a setting_ and therefore they are attempting to do something specific and we shouldn't overwrite that `via .token ||= ENV['']` for example and raise an error. The user example is someone writing `config.token = ENV['DOES_NOT_EXIST']` and somehow defaulting to `ENV['SLACK_API_TOKEN']` magically would be a bug.

Reply:
I’m thinking I’ll keep the existing code for aliases in this PR for now (this was from copy paste), will be cleaner to look at it separately :) is that ok?

As for the token, didn’t think of that - Does this mean we default to having the ENV variables take precedence? (e.g. replace this only `if ENV.key?`) Else I’m a little uncertain how to handle it without a `||=`.

Reply:
Keep existing: yes.

ENV doesn't take precedence, all I am saying is that there's a difference between `ENV['X']` and `ENV.key?('X')`. The former returns `false` when `ENV['X'] == nil`, but the latter does not. We want the latter.

Reply:
👌 got it, though technically if both were present, the ENV would take precedence unless they dive into the App instance